src/cli/rustup_mode.rs

```diff
@@ -55,7 +55,16 @@ where
 pub fn main() -> Result<utils::ExitCode> {
     self_update::cleanup_self_updater()?;
 
-    let matches = cli().get_matches_from(process().args_os());
+    let matches = match cli().get_matches_from_safe(process().args_os()) {
+        Ok(matches) => Ok(matches),
+        Err(e)
+            if e.kind == clap::ErrorKind::HelpDisplayed
+                || e.kind == clap::ErrorKind::VersionDisplayed =>
+        {
+            return Ok(utils::ExitCode(0))
+        }
+        Err(e) => Err(e),
+    }?;
```

Comment:
It might be nice to have this factored into a function given it's also used in setup_mode.rs and someone fixing something in one might miss the other.